sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -0,0 +1,121 @@
+        /// </summary>
+        internal ReadOnlyMemory<byte> Memory { get; }
+
+        /// <summary>
+        /// Creates a binary data instance from bytes.
+        /// </summary>
+        /// <param name="data">Byte data.</param>
+        public BinaryData(ReadOnlyMemory<byte> data)
+        {
+            Memory = data;
+        }
+
+        /// <summary>
+        /// Creates a binary data instance from a string
+        /// using UTF-8 encoding.
+        /// </summary>
+        /// <param name="data">The string data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static BinaryData Create(string data) =>
+            Create(data, Encoding.UTF8);
+
+        /// <summary>
+        /// Creates a binary data instance from a string
+        /// using the specified encoding.
+        /// </summary>
+        /// <param name="data">The string data.</param>
+        /// <param name="encoding">The encoding to use when converting
+        /// the data to bytes.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static BinaryData Create(string data, Encoding encoding)
+        {
+            Argument.AssertNotNull(encoding, nameof(encoding));
+            return new BinaryData(encoding.GetBytes(data));
+        }
+
+        /// <summary>
+        /// Creates a binary data instance from the specified stream.
+        /// </summary>
+        /// <param name="stream">Stream containing the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static BinaryData Create(Stream stream)
+        {
+            Argument.AssertNotNull(stream, nameof(stream));
+            using (var memoryStream = new MemoryStream())
+            {
+                stream.CopyTo(memoryStream);
+                return new BinaryData(memoryStream.ToArray());
+            }
+        }
+
+        /// <summary>
+        /// Creates a BinaryData instance from the specified data using
+        /// the specified type.
+        /// </summary>
+        /// <typeparam name="T">The type of the data.</typeparam>
+        /// <param name="data">The data to use.</param>
+        /// <param name="serializer">The serializer to serialize
+        /// the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static BinaryData Create<T>(
+            T data,
+            ObjectSerializer serializer)
+        {
+            Argument.AssertNotNull(serializer, nameof(serializer));
+            using var memoryStream = new MemoryStream();
+            serializer.Serialize(memoryStream, data, typeof(T));
+            memoryStream.Position = 0;
+            return Create(memoryStream);
+        }
+
+        /// <summary>
+        /// Implicit conversion to string.
+        /// </summary>
+        /// <param name="data"></param>
+        public static implicit operator string(BinaryData data) =>
+            data.AsString();
```

Comment:
Need to worry about nulls if this is a class, here and on the rest.

Reply:
Not a class anymore